src/components/MeasurementInfoPanel.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+
+import { ValueRenderers } from '..';
+
+import { Table } from './Table';
+import { MeasurementBase } from './context/data/DataState';
+
+export interface MeasurementInfoPanelProps {
+  measurement: MeasurementBase;
+}
+
+export function MeasurementInfoPanel(props: MeasurementInfoPanelProps) {
+  const {
+    measurement: { meta, info },
+  } = props;
+
+  const [search, setSearch] = useState('');
+  const metaRows = (search: string) =>
+    Object.keys(meta).map((key) => {
+      const value = meta[key];
+      if (
+        !key.toLowerCase().includes(search.toLowerCase()) &&
+        !valueSearch(value, search)
+      ) {
+        return null;
+      }
+
+      return (
+        <Table.Row key={key}>
+          <ValueRenderers.Text value={key} />
+          {valueCell(value)}
+        </Table.Row>
+      );
+    });
+  const infoRows = (search: string) =>
```

Comment:
Please avoid repeating the same code twice and add a function.

Reply:
fixed thank you